src/nvim/os/unix_defs.h

```diff
@@ -31,12 +31,6 @@
 #ifndef USR_EXRC_FILE
 # define USR_EXRC_FILE "~/.exrc"
 #endif
-#ifndef USR_VIMRC_FILE
-# define USR_VIMRC_FILE "~/.nvimrc"
```

Comment:
At any rate this was already decided. Use the terrible default locations and don't support old vimrc locations. 

What if addd a command line option like `nvim --legacyconfig` which automatically sets `XDG*` env vars to match the old vimrc locations? So at least users who don't want to change only need to know this one option, and internally all the code stays the same (except for the function that sets the `XDG*` variables).